apps/src/applab/designMode.js

```diff
@@ -1216,6 +1218,36 @@ designMode.parseFromLevelHtml = function (rootEl, allowDragging, prefix) {
   children.appendTo(rootEl);
 };
 
+// We have seen issues where design mode HTML somehow contains HTML from
+// outside the design mode area (ex. codeApp, visualizationArea). We don't know
+// exactly how this is happening, but this function removes those selectors to
+// help prevent the page from breaking.
+function sanitizeLevelDom(levelDom) {
+  const disallowedSelectors = [
```

Comment:
Could you explain how you know this list is complete? [Here](https://codedotorg.slack.com/files/U03J6PY3014/F04V234L0RH/main__1_.json?origin_team=T039SAH7W&origin_channel=C03DBDN67B7) is a past `main.json` with the added html ([Slack thread](https://codedotorg.slack.com/archives/C03DBDN67B7/p1679442847568599)). Would this list take care of all the extra html in this file?

Reply:
Yep, I'm going off the previous projects we've seen. `#codeApp` contains most of the content in the lab, so anytime that is included, removing it will remove any other offending HTML. In a few of the other projects though, we saw that `#codeApp` wasn't present, but its immediate children were, which include `#visualizationColumn`, `#visualizationResizeBar` and `.editor-column`. 

Thanks for the link! This code worked on the example you linked - the outermost container of the extraneous HTML was a `#codeApp` div, so removing those removed all the extra HTML.

Reply:
Thanks for explaining and for checking the eample file. Such a cool solution.